### ISSUE TITLE
Using ^C to stop mp4 recording corrupts file (windows)

### ISSUE THREAD

**Issue by evillurker** [Genymobile/scrcpy#5373]
- **OS:** [Windows.]
 - **Scrcpy version:** [2.7]
 - **Installation method:** [windows release]
 - **Device model:** [Galaxy S23]
 - **Android version:** [14]

## Describe the bug

When finishing with ^C recording, it says it completed but the file is corrupt when choosing mp4 method. mkv doesn't produce this bug but it produces jagged quality in the first few seconds. I tried using adb to kill the process but it keeps recording when trying to do that.

**Comment by rom1v** [comment]
Do you use "Windows Terminal"? https://github.com/Genymobile/scrcpy/issues/5122#issuecomment-2265289619

> I tried using adb to kill the process but it keeps recording when trying to do that.

That's should not be possible. if you kill adb, then scrcpy terminates, and this terminates the recording.

Please post a whole console output when you kill adb (also include the command you executed).

Does it work correctly when you close the scrcpy window (not the terminal)?

**Comment by evillurker** [comment]
Nevermind, seems like all I had to do is taskkill adb.exe

**Comment by rom1v** [comment]
But the best is to close the scrcpy window.

Ctrl+c should also work in theory, but not with Windows terminal for some reason.

**Comment by evillurker** [comment]
Ctrl C in the console window works 50% of the time, the other 50% in windows it corrupts the header of the mp4. I am waiting for the --time-limit on your next version, until then I have to make my own GUI to stop recording

**Comment by rom1v** [comment]
> Ctrl+c should also work in theory, but not with Windows terminal for some reason.

> Ctrl C in the console window works 50% of the time

When I say that it does not work, I mean that it abruptly kill the process (which is never ok), instead of sending the equivalent of SIGINT so that the process terminates gracefully and scrcpy can finish running properly.

**Comment by rom1v** [comment]
> I am waiting for the --time-limit on your next version

You can use the binary from #5370, it includes the time-limit fix (afbaf59abba79a79ab6f4c659ff3d832e02a8e7f).

**Comment by evillurker** [comment]
> > I am waiting for the --time-limit on your next version
> 
> You can use the binary from #5370, it includes the time-limit fix ([afbaf59](https://github.com/Genymobile/scrcpy/commit/afbaf59abba79a79ab6f4c659ff3d832e02a8e7f)).

Thank you, I didn't know I can download it. This will be very useful.

**Comment by rom1v** [comment]
Please also test interrupting using Ctrl+c with this version: https://github.com/Genymobile/scrcpy/issues/5122#issuecomment-2408888597

That will help to fix the problem.

**Comment by evillurker** [comment]
Ctrl C does not always work, I try using it and sometimes it reacts, sometimes it just keeps recording until i manually close the window. Last time I closed the window the file was corrupted. But the time--limit works perfectly.

**Comment by rom1v** [comment]
> Ctrl C does not always work

Even with the version I published this morning here? https://github.com/Genymobile/scrcpy/issues/5122#issuecomment-2408888597 (on that version, time-limit will still be broken though)

**Comment by evillurker** [comment]
> > I am waiting for the --time-limit on your next version
> 
> You can use the binary from #5370, it includes the time-limit fix ([afbaf59](https://github.com/Genymobile/scrcpy/commit/afbaf59abba79a79ab6f4c659ff3d832e02a8e7f)).

With this version

**Comment by rom1v** [comment]
> With this version

No, as a test to fix recording interruption on Windows, I'd like you to test https://github.com/Genymobile/scrcpy/issues/5122#issuecomment-2408888597 please.

**Comment by evillurker** [comment]
Ok i'll give it a test soon when I'm done recording some stuff that needs time-limit

**Comment by evillurker** [comment]
As of now after testing, ^C works fine when directly running scrcpy.exe, but still corrupts if used in console window through cmd.exe sometimes. So basically it needs to be directly running and not through a cmd window for some reason.

**Comment by rom1v** [comment]
Thank you for your tests. Is your terminal "Windows terminal" (like in the videos in https://github.com/Genymobile/scrcpy/issues/5122#issuecomment-2242503206), or the old default console (https://github.com/Genymobile/scrcpy/issues/5122#issuecomment-2265153630)?

**Comment by evillurker** [comment]
It's not Windows Terminal, it's cmd.exe
![image](https://github.com/user-attachments/assets/61cbdfcc-49ac-42bd-a737-6de40f1b9310)

**Comment by rom1v** [comment]
OK. Do you get the line `Recording complete to mp4 file: file.mp4` at the end when you press Ctrl+c after running `scrcpy --record=file.mp4`?

**Comment by evillurker** [comment]
Yes, I get that line, but 50% of the time (maybe more) its corrupted, unplayable
Does not happen with mkv however the mkv has jagged/scrambled display on the first few seconds.

**Comment by rom1v** [comment]
> Yes, I get that line

Oh ok, that's.a.diffeeent problem then.

Could you please post such a corrupted file?

**Comment by evillurker** [comment]
https://github.com/user-attachments/assets/2ef66425-dc4b-4d06-bd04-983a723a6c40

**Comment by rom1v** [comment]
OK, that's very strange if `Recording complete to mp4 file: file.mp4` is correctly printed.

Can you reproduce the problem if you close the scrcpy window?

**Comment by evillurker** [comment]
It depends on if I run it through cmd or directly through the .exe
What do you want to test? Maybe just contact me on discord so we can test it live, if you want. EvilLurker

**Comment by mjzhaochenyi** [comment]
I am using it on Mac, and the file is always corrupt if I record a long video. The short video is fine.

**Comment by rom1v** [comment]
How do you stop recording on your mac? DO NOT close the console, but either close the scrcpy window or Ctrl+c in the console.

**Comment by mjzhaochenyi** [comment]
> How do you stop recording on your mac? DO NOT close the console (i.e. do not abruptly kill the app), but either close the scrcpy window or Ctrl+c in the console.

I tried "close app window" and "ctrl+c", but the results were the same. My long videos are normally 3 hours sessions.

**Comment by rom1v** [comment]
What's the output in the console when you close scrcpy? (in particular the log about recording)
